File: app/src/main/java/com/happs/ximand/ringcontrol/RingControlApplication.java
package com.happs.ximand.ringcontrol;

import android.app.Application;

import com.happs.ximand.ringcontrol.model.dao.SharedPreferencesDao;
import com.happs.ximand.ringcontrol.model.repository.impl.TimetableRepository;

public class RingControlApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        TimetableRepository.initialize(this);
        SharedPreferencesDao.initialize(this);
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
    }
}

File: app/src/main/java/com/happs/ximand/ringcontrol/RingControlApplication.kt
package com.happs.ximand.ringcontrol

import android.app.Application
import com.happs.ximand.ringcontrol.model.dao.SharedPreferencesDao
import com.happs.ximand.ringcontrol.model.repository.impl.TimetableRepository

class RingControlApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        TimetableRepository.initialize(this)
        SharedPreferencesDao.initialize(this)
    }

}